aws-blog-event-driven-batch-analytics/src/main/java/com/amazonaws/bigdatablog/edba/LambdaContainer.java

```diff
@@ -0,0 +1,249 @@
+		ResultSet rs = conditionFetchStmt.executeQuery(conditionFetchSQL);
+		PreparedStatement updateJobConfigPS = conn.prepareStatement(updateJobConfigStatusSQL);
+		Statement jobInputFilesMinTimestampStmt = conn.createStatement();
+		Statement updateSubmittedJobsStmt=conn.createStatement();
+		List<String> activeClusters = getActiveClusterId();
+		String clusterId = null;
+		while(rs.next()){
+			ResultSet conditionQueryResult = jobInputFilesMinTimestampStmt.executeQuery(jobInputFilesMinTSAndCountSQL+rs.getString("job_input_pattern"));
+			conditionQueryResult.next();
+			if(conditionQueryResult.getTimestamp("min_lvt").after(rs.getTimestamp("last_run_timestamp")) 
+					&&
+					conditionQueryResult.getInt("file_count") >= rs.getInt("job_min_file_count")
+					&&
+					isAdditionalCriteriaPassed(rs.getString("job_addl_criteria"),conn)){
+				clusterId = activeClusters.get(new Random().nextInt(activeClusters.size()-0));
+				String jobId = fireEMRJob(rs.getString("job_params"),clusterId);
+				updateJobConfigPS.setString(1,clusterId+":"+jobId);
+				updateJobConfigPS.setString(2, rs.getString("job_config_id"));
+				updateJobConfigPS.addBatch();
+				updateSubmittedJobsStmt.addBatch(updateSubmittedJobsJSON.replaceAll("\\?", rs.getString("job_config_id"))+rs.getString("job_input_pattern"));
+
+			}
+		}
+		updateJobConfigPS.executeBatch();
+		updateSubmittedJobsStmt.executeBatch();
+		updateSubmittedJobsStmt.close();
+		updateJobConfigPS.close();
+		conditionFetchStmt.close();
+		conn.close();
+	}
+	
+	// EMR Job Monitor lambda function
+	
+	public void monitorEMRStep() throws Exception {
+//		Class.forName("com.mysql.jdbc.Driver").newInstance();
+		List<String> stepIds = new ArrayList<String>();
+		Connection conn = DriverManager.getConnection(url, user, password);
+		ResultSet openStepsRS = conn.createStatement().executeQuery(retrieveOpenSteps);
+		AmazonElasticMapReduceClient emr = new AmazonElasticMapReduceClient();
+		DescribeStepRequest stepReq=new  DescribeStepRequest();
+		PreparedStatement ps = conn.prepareStatement(updateStepStatus);
+		while(openStepsRS.next()){
+			
+			stepReq.setClusterId(openStepsRS.getString("cluster_id"));
+			stepReq.setStepId(openStepsRS.getString("step_id"));
+			String stepState = emr.describeStep(stepReq).getStep().getStatus().getState();
+			
+				if(stepState.equals(StepState.COMPLETED.toString())){
+					ps.setString(1,StepState.COMPLETED.toString());
+				}else if (stepState.equals(StepState.FAILED.toString())){
+					ps.setString(1,StepState.FAILED.toString());					
+				}
+				ps.setString(2,openStepsRS.getString("job_config_id"));
+				ps.addBatch();
+		}
+		
+		ps.executeBatch();
+		ps.close();
+		conn.close();
+	}
+	
+	// adds EMR step
+	protected String fireEMRJob(String paramsStr,String clusterId){
+		StepFactory stepFactory = new StepFactory();
+		AmazonElasticMapReduceClient emr = new AmazonElasticMapReduceClient();
+		emr.setRegion(Region.getRegion(Regions.US_EAST_1));
```

Comment:
This should establish the region from the Lambda environment. Use `System.getEnv().get('AWS_REGION')`.